### PR TITLE
introduce Mac-specific shims for `utimenstat()` and for `stat.st_mtim`

### LINKED ISSUE #768: make compiler issues on os x sierra
```
Karens-MacBook-Pro:SFTtech-openage-db976c0` thomas$ make
[  0%] Automatic moc for target libopenage
Scanning dependencies of target libopenage
[  0%] Building CXX object libopenage/CMakeFiles/libopenage.dir/assetmanager.cpp.o
In file included from /Users/thomas/Downloads/SFTtech-openage-db976c0/libopenage/assetmanager.cpp:16:
In file included from /Users/thomas/Downloads/SFTtech-openage-db976c0/libopenage/texture.h:12:
In file included from /Users/thomas/Downloads/SFTtech-openage-db976c0/libopenage/coord/tile.h:9:
/Users/thomas/Downloads/SFTtech-openage-db976c0/libopenage/coord/../util/misc.h:146:25: error: constexpr function never produces a constant expression [-Winvalid-constexpr]
inline constexpr size_t array64_size(size_t count) {
                        ^
/Users/thomas/Downloads/SFTtech-openage-db976c0/libopenage/coord/../util/misc.h:147:9: note: non-constexpr function 'ceil' cannot be used in a constant expression
        return std::ceil(static_cast<double>(count) / static_cast<double>(uint64_s));
               ^
/usr/include/math.h:470:15: note: declared here
extern double ceil(double);
              ^
1 error generated.
make[3]: *** [libopenage/CMakeFiles/libopenage.dir/assetmanager.cpp.o] Error 1
make[2]: *** [libopenage/CMakeFiles/libopenage.dir/all] Error 2
make[1]: *** [all] Error 2
make: *** [build] Error 2
Karens-MacBook-Pro:SFTtech-openage-db976c0 thomas$ 
```

### PR DESCRIPTION
**Pull request by Birch-san**
`utimenstat()` interprets nullptr as "use current time", so I used POSIX `clock_gettime()` to grab current time.

Apparently [`utimes()` can be used instead of `utimensat()`](https://github.com/mpartel/bindfs/issues/6#issuecomment-42538754), we just needed to convert timespec into a timeval.

Fixed the `st_mtim` problem also. We wanted a "modified timestamp" of type `struct timespec`. [`st_mtimespec` should do the trick](https://gist.github.com/Birch-san/ebca41072b8ade2be0bfbc18b110fb94#file-stat_2_sierra-txt-L79).

Fixes #768

### CODE REVIEW

File: libopenage/util/fslike/directory.cpp [hunk 1]
```diff
@@ -202,8 +207,21 @@ bool Directory::touch(const Path::parts_t &parts) {
...
+	// update the timestamp	
+#ifdef __APPLE__
+    struct timespec ts[2];
+    clock_gettime(CLOCK_REALTIME, ts);
+    struct timeval tv[2];
+    tv[0].tv_sec = ts[0].tv_sec;
+    tv[0].tv_usec = ts[0].tv_nsec*1000;
+    tv[1].tv_sec = ts[1].tv_sec;
+    tv[1].tv_usec = ts[1].tv_nsec*1000;
+    int result = utimes(path.c_str(), tv);
```
Comment (TheJJ) [inline libopenage/util/fslike/directory.cpp]: tab indent pls

File: libopenage/util/fslike/directory.cpp [hunk 2]
```diff
@@ -202,8 +207,21 @@ bool Directory::touch(const Path::parts_t &parts) {
...
+#else
+    int result = utimensat(AT_FDCWD, path.c_str(), nullptr, 0) == 0;
```
Comment (TheJJ) [inline libopenage/util/fslike/directory.cpp]: here as well

File: libopenage/util/fslike/directory.cpp [hunk 3]
```diff
@@ -202,8 +207,21 @@ bool Directory::touch(const Path::parts_t &parts) {
 
 	close(fd);
 
-	// update the timestamp
-	return utimensat(AT_FDCWD, path.c_str(), nullptr, 0) == 0;
+	// update the timestamp
```
Comment (TheJJ) [inline libopenage/util/fslike/directory.cpp]: here's the trailing whitespace kevin complains about

File: libopenage/util/fslike/directory.cpp (hunk 2 above)
Comment (Birch-san) [inline libopenage/util/fslike/directory.cpp]: done

File: libopenage/util/fslike/directory.cpp (hunk 1 above)
Comment (Birch-san) [inline libopenage/util/fslike/directory.cpp]: done

**Comment by TheJJ** [comment]
nice, if you could now please [rebase and squash](https://github.com/SFTtech/openage/blob/master/doc/contributing.md#rebasing) those commits into one, we can ship it.

**Comment by billforward-alex** [comment]
regarding `copying.md`: is there a way for me to write my email address in a less machine-parseable way? (or omitting it / providing other instructions on how to contact)? I don't like the idea of my email address's being extremely well-indexed.

**Comment by janisozaur** [comment]
@Birch-san I ran into the same issue, I used gmail's aliases to alleviate this problem a little. But as long as it is an email registered with GitHub, it should all work, so in theory you should be able to use 10minutemail just as well, or http://www.fakenamegenerator.com/ for something a little more permanent, yet temporary.

**Comment by TheJJ** [comment]
Good point, maybe we should use one of those standard techniques to obfuscate the emails a bit.

**Comment by Seadragon91** [comment]
Maybe this can help, too.
https://help.github.com/articles/keeping-your-email-address-private/

**Comment by TheJJ** [comment]
Unfortunately not really, as we want to have real email addresses to reach contributors someday, e.g. if we ever had licensing issues.

**Comment by billforward-alex** [comment]
presumably if I ensure that `Birch-san@users.noreply.github.com` forwards to my actual email address, everything should be fine?

**Comment by billforward-alex** [comment]
not sure whether it supports forwarding…

**Comment by TheJJ** [comment]
Nah it's more about that we 0. don't wanna depend on github and 1. real emails are less likely to be abandoned by users. We could just update the `copying.md` to save mails as `accountname %0x40 host` :)

**Comment by billforward-alex** [comment]
taken to logical extreme:

- generate private key, distribute to a few key owners and to CI system
- save public key into repo
- encrypt contributor email addresses with public key
- CI system has private key and so can read encrypted email addresses from `copying.md`

but then I think "well, all our commits are already tagged with an email anyway… the main thing we want to protect against is being _easily_ indexed by bots". so maybe something closer to `accountname %0x40 host` would be fine.

**Comment by TheJJ** [comment]
Jup, lets not overcomplicate things. I'd just obfuscate a bit, and we should be good (or at least better than before). Do you mind creating a separate pull request for that?

**Comment by billforward-alex** [comment]
I'll see what I can do.

**Comment by Birch-san** [comment]
accidentally posted all of the above comments from my work account. I'm gonna delete them so that only my personal account is involved

> regarding `copying.md`: is it possible for me to provide my email address in a less machine-parseable way? (or to omit it / provide other instructions on how to contact)? I don't like the idea of my email address's being extremely well-indexed.

> presumably if I ensure that `Birch-san@users.noreply.github.com` forwards to my actual email address, everything should be fine?

> not sure whether it supports forwarding…

> taken to logical extreme:
> 
> - generate private key, distribute to a few key owners and to CI system
> - save public key into repo
> - encrypt contributor email addresses with public key
> - CI system has private key and so can read encrypted email addresses from copying.md
> 
> but then I think "well, all our commits are already tagged with an email anyway… the main thing we want to protect against is being easily indexed by bots". so maybe something closer to accountname %0x40 host would be fine.

> I'll see what I can do.

**Comment by Birch-san** [comment]
Let's get email obfuscation shipped before I join the contributors list:
https://github.com/SFTtech/openage/pull/779

**Comment by TheJJ** [comment]
Mail obfuscation is live!

**Comment by Birch-san** [comment]
Rebased onto master, resolved conflict on `copying.md`, squashed, amended author & committer, resigned commit

**Comment by TheJJ** [comment]
Thank you, well done :)

**Pull request merged**